module/output/ImageCompressor/src/ImageCompressor.cpp

```diff
@@ -12,175 +12,168 @@
-                        c["concurrent"].as<int>());
-                }
-                else if (c["name"].as<std::string>() == "turbojpeg") {
-                    config.factories.emplace_back(
-                        std::make_shared<compressor::turbojpeg::Factory>(c["quality"].as<int>()),
-                        c["concurrent"].as<int>());
-                }
-            }
-        });
-
-        on<Trigger<Image>>().then("Compress Image", [this](const Image& image) {
-            // Find our list of compressors, and if needed recreate it
-            std::shared_ptr<CompressorContext> ctx;
-
-            /* Mutex Scope */ {
-                std::lock_guard<std::mutex> lock(compressor_mutex);
-                auto it = compressors.find(image.camera_id);
-                if (it == compressors.end() || it->second->width != image.dimensions[0]
-                    || it->second->height != image.dimensions[1] || it->second->format != image.format) {
-                    log<NUClear::INFO>("Rebuilding compressors for", image.name, "camera");
-
-                    // Replace the existing one with a new one
-                    it = compressors.insert(std::make_pair(image.camera_id, std::make_shared<CompressorContext>()))
-                             .first;
-                    it->second->width  = image.dimensions[0];
-                    it->second->height = image.dimensions[1];
-                    it->second->format = image.format;
-
-                    for (auto& f : config.factories) {
-                        for (int i = 0; i < f.second; ++i) {
-                            it->second->compressors.emplace_back(CompressorContext::Compressor{
-                                std::make_unique<std::atomic<bool>>(),
-                                f.first->make_compressor(image.dimensions[0], image.dimensions[1], image.format),
-                            });
-                        }
+            auto it = compressors.find(image.id);
+            if (it == compressors.end() || it->second->width != image.dimensions[0]
+                || it->second->height != image.dimensions[1] || it->second->format != image.format) {
+                log<NUClear::INFO>("Rebuilding compressors for", image.name, "camera");
+
+                // Replace the existing one with a new one
+                it = compressors.insert(std::make_pair(image.id, std::make_shared<CompressorContext>())).first;
+                it->second->width  = image.dimensions[0];
+                it->second->height = image.dimensions[1];
+                it->second->format = image.format;
+
+                for (auto& f : config.factories) {
+                    for (int i = 0; i < f.second; ++i) {
+                        auto a = std::make_unique<std::atomic<bool>>();
+                        it->second->compressors.emplace_back(CompressorContext::Compressor{
+                            std::move(a),
```

Comment:
Same as before.

Reply:
Same as before, clang static analysis thinks it's a memory leak if you don't